### ISSUE TITLE
`Style/ClassAndModuleChildren` does not handle comments properly

### ISSUE THREAD

**Issue by dvandersluis** [rubocop/rubocop#9767]
### `EnforcedStyle: compact`

```ruby
class A # class A comment
  class B # class B comment
  end
end
```

autocorrects to

```ruby
class A::B # class B comment
end
```

The `class A comment` disappears.

---

### `EnforcedStyle: nested`

```ruby
class A::B # class A comment
end
```

autocorrects to

```ruby
module A
  class B # class A comment
  end
end
```

The comment moves to the inner class which may not be desired.

## RuboCop version

`master`

**Comment by dvandersluis** [comment]
I think in both cases, we should probably just move all comments to above the outer class/module.

**Comment by dvandersluis** [comment]
@tejasbubane I actually have never even noticed that `Style/InlineComment` exists! It's disabled by default.

But that's not actually what I was getting at. My point was that there's no way of knowing if the inline comment was for module A or class B in the original case, so putting it on top is probably "safer".

**Comment by tejasbubane** [comment]
> My point was that there's no way of knowing if the inline comment was for module A or class B in the original case, so putting it on top is probably "safer".

```ruby
class A::B # some comment
```
Since this is definition of `B`, doesn't that mean the comment if for `B` since `A` is just a namespace wrapper?

My point is `EnforcedStyle: compact` for this cop actually eats up the comment which is definitely a bug. 
`EnforcedStyle: nested` on the other hand, leaves the comment as is and I feel it is not the job of this cop to move it up.
